sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -439,6 +439,39 @@ object StateStoreProvider {
   }
 }
 
+/**
+ * This is an optional trait to be implemented by [[StateStoreProvider]]s that can read fine
+ * grained state data which is replayed from a specific snapshot version. It is used by the
+ * snapshotStartBatchId option in state data source.
+ */
+trait FineGrainedStateSource {
```

Comment:
not sure that this is the best name for the trait here.

how about just calling it `StoreReplay` or similar ?

Reply:
Let's rename it to `SupportsFineGrainedReplayFromSnapshot`.